clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -448,6 +466,7 @@ private static int parseAcks(String acksString) {
         try {
             // first make sure the metadata for the topic is available
             ClusterAndWaitTime clusterAndWaitTime = waitOnMetadata(record.topic(), record.partition(), maxBlockTimeMs);
+            waitOnPid(maxBlockTimeMs);
```

Comment:
I think this should be kept separate from `waitOnMetadata`. But yes, we should use the `remainingWaitTime` after `waitOnMetadata` completes, and also update that value after this call is made. Will update the patch.